--secrets "keyVaultSecret" when using az vm create

WiseMack opened the pull request:
Line 23 added _validate_secrets at importer
Line 173 added secrets parameter
Line 222-223 Left the original and added new commented secrets parameter below
Line 416-417 Left the original and added new commented secrets parameter below

The Secrets parameter importer below was created by using the examples in Lines 222 and 416, let's go step by step:

c.argument('secrets', multi_ids_type, options_list=['--secrets', '-s'], help='One or many Key Vault secrets as JSON strings or files via `@{path}` containing `[{ "sourceVault": { "id": "value" }, "vaultCertificates": [{ "certificateUrl": "value", "certificateStore": "cert store name (only on windows)"}] }]`',  validator=_validate_secrets)

'secrets' the argument name
multi_ids_type --> No idea what it does, but 222 and 416 both had it.
options_list=['--secrets', '-s'] the optional name for the parameter
help='One or many Key Vault secrets as JSON strings or files via `@{path}` containing `[{ "sourceVault": { "id": "value" }, "vaultCertificates": [{ "certificateUrl": "value", "certificateStore": "cert store name (only on windows)"}] }]`' The help!
type=file_type No idea what this is but unlike multi_ids_type 222 didn't have it and 416 did, so I decided to remove it
completer=FilesCompleter()  No idea what this is but unlike multi_ids_type 222 didn't have it and 416 did, so I decided to remove it
validator=_validate_secrets The ONLY thing I added

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

WiseMack commented:
@tjprescott there seem to be a misunderstanding, the commented lines were added wholesale, I did not change the original code at all, I did not remove any logic.

tjprescott commented:
@WiseMack ah, you're right. If the PR is still in progress, I'd recommend reopening later when you feel it is ready. As written, it is a no-op.

WiseMack commented:
It is not in progress, I simply left it in comments in order to make it stant out more
1. In line 23 The function "_validate_secrets" despite being part of azure.cli.command_modules.vm._validators  is neither imported nor used in the original file.
The function allows the validation of a secret's format, it is not among the imports and of course is not used in the _params.py file.
2. In line 162 there is a function
with self.argument_context('vm create') as c:
this vm create does NOT have a --secrets argument, I added it whole sale.
3. In line 220 there is another function
`with self.argument_context('vm secret') as c:`
There is an argument --secrets BUT
Original argument:
`c.argument('secrets', multi_ids_type, options_list=['--secrets', '-s'], help='Space-separated list of key vault secret URIs. Perhaps, produced by \'az keyvault secret list-versions --vault-name vaultname -n cert1 --query "[?attributes.enabled].id" -o tsv\'')`
New argument
`c.argument('secrets', multi_ids_type, options_list=['--secrets', '-s'], help='One or many Key Vault secrets as JSON strings or files via `@{path}` containing `[{ "sourceVault": { "id": "value" }, "vaultCertificates": [{ "certificateUrl": "value", "certificateStore": "cert store name (only on windows)"}] }]`',  validator=_validate_secrets)
`
Differences:
a. the help is different, I found the help in line 415 to be EASIER to understand, so I changed it.
b. I added validator=_validate_secrets (The original didn't have it)

4. In line 406
There is a function
 for scope in ['vm create', 'vmss create']:

Original argument:
`c.argument('secrets', multi_ids_type, help='One or many Key Vault secrets as JSON strings or files via `@{path}` containing `[{ "sourceVault": { "id": "value" }, "vaultCertificates": [{ "certificateUrl": "value", "certificateStore": "cert store name (only on windows)"}] }]`', type=file_type, completer=FilesCompleter())`
New argument
`c.argument('secrets', multi_ids_type, options_list=['--secrets', '-s'], help='One or many Key Vault secrets as JSON strings or files via `@{path}` containing `[{ "sourceVault": { "id": "value" }, "vaultCertificates": [{ "certificateUrl": "value", "certificateStore": "cert store name (only on windows)"}] }]`',  validator=_validate_secrets)`

Differences:
a. I removed the parametes type=file_type AND completer=FilesCompleter()) because I believe they do not apply to the --secrets parameter
b. I added validator=_validate_secrets (The original didn't have it)

WiseMack commented:
@tjprescott It seems my last message didn't notify you.

tjprescott commented:
The PR in GitHub shows one file changed (params.py) and only four lines changed in that one file. In all cases, those four lines are comments.
